### PR TITLE
Per-stream metrics

### LINKED ISSUE #2172: metrics: start exporting stream-specific metrics everywhere

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
Adds per-stream metrics. Also adds Broadcaster's IP to Orchestrator's metric.

**Specific updates (required)**
- Added `manifest_id` label to some metrics
- Expose `mil_pixels_processed` metric on Orchestrator
- For `mil_pixels_processed` metric added `client_ip` label which represent Broadcaster's IP address
- Expose Eth address in payments metrics

All those are opt-it: there is two new command-line switches that off by default.

**How did you test each of these updates (required)**
Manually

**Does this pull request close any open issues?**
Fixes #2172 

**Checklist:**

- [ ] Read the [contribution guide](./doc/contributing.md)
- [ ] `make` runs successfully
- [ ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

### CODE REVIEW

File: clog/clog.go [hunk 1]
```diff
@@ -98,6 +100,22 @@ func AddVal(ctx context.Context, key, val string) context.Context {
 	return ctx
 }
 
+func GetManifestID(ctx context.Context) string {
+	return GetVal(ctx, manifestID)
+}
+
+func GetVal(ctx context.Context, key string) string {
```
Comment (figintern) [inline clog/clog.go:107]: Where/what is this Getter used for, is it related to the PR change?

Comment (darkdarkdragon) [reply clog/clog.go:107]: https://github.com/livepeer/go-livepeer/blob/bf775af0a95e58960732b5283257f32447c3ea03/monitor/census.go#L872

File: core/orchestrator.go [hunk 2]
```diff
@@ -203,22 +203,22 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
...
 				}
 			}(ticket, tsp.Sig, seed)
 		}
 	}
 
 	if monitor.Enabled {
-		monitor.TicketValueRecv(totalEV)
-		monitor.TicketsRecv(totalTickets)
-		monitor.WinningTicketsRecv(totalWinningTickets)
+		monitor.TicketValueRecv(ctx, sender.Hex(), totalEV)
```
Comment (figintern) [inline core/orchestrator.go:219]: 👍

File: monitor/census.go [hunk 3]
```diff
@@ -185,12 +192,13 @@ type (
 	}
 
 	segmentsAverager struct {
-		segments  []segmentCount
-		start     int
-		end       int
-		removed   bool
-		removedAt time.Time
-		tries     map[uint64]tryData // seqNo:try
+		manifestID string
```
Comment (figintern) [inline monitor/census.go:195]: What is `segmentsAverager` used for? Does adding manifestID to it affect the dimensionality, want to confirm if it's now creating an `averager` per stream as opposed to before per lines `1451`, `1468` in `census.go`, will this affect the functionality or how it's used?

**Comment by figintern** [comment]
Nice! Trying to grok why Context is required/passed everywhere now, is that where `manifestID` is stored?

File: monitor/census.go (hunk 3 above)
Comment (darkdarkdragon) [reply monitor/census.go:195]: `averager` is created per-stream, before and after the change

**Comment by darkdarkdragon** [comment]
@tqian1 
> Nice! Trying to grok why Context is required/passed everywhere now, is that where manifestID is stored?

Yes, context holds some key info we're including in log messages, including manifestID

File: monitor/census.go (hunk 3 above)
Comment (figintern) [reply monitor/census.go:195]: So just including the ID which wasn't present before, got it

**Review by figintern — APPROVED** [review]
LGTM! Any testing/metrics to look at on staging?

**Pull request merged**